### ISSUE TITLE
Passthrough mode not working

### ISSUE THREAD

**Issue by nhan000** [jglev/obsidian-paste-mode#31]
Previously, when set paste mode to Passthrough, I was able to paste text by pressing `Ctrl+V` twice but not images (as said in https://github.com/jglev/obsidian-paste-mode/issues/30). However, now I can't even paste any text in Passthrough mode. Other modes seem to work fine.

I've tried this in a copied sandbox vault and have restarted Obsidian multiple times to test it.

Obsidian v1.3.5, Windows 11, paste mode v5.0.2

**Comment by LoneFireBlossom** [comment]
same problem, so now I change to "plain text" mode and it works well

**Comment by jasonshelter0** [comment]
Yes, and I asked codex, here's a temporary solution working for me:

Modify the main.js:

  Before (main.js:4336–4344)

    this.app.workspace.on("editor-paste", (evt, editor) => __async(this, null, function* () {
    if (evt.defaultPrevented) {
    return;
    }
    evt.preventDefault();
    let mode = this.settings.mode;
    return;
    }

  After (swap order so default paste is not blocked in Passthrough)

    this.app.workspace.on("editor-paste", (evt, editor) => __async(this, null, function* () {
    if (evt.defaultPrevented) {
    return;
    }
    let mode = this.settings.mode;
    if (mode === Mode.Passthrough) {
    return; // allow native/other-handlers paste to proceed
    }
    evt.preventDefault();

**Comment by jasonshelter0** [comment]
Disable & re-enable the plugin, or restart obsidian to make it work.

**Comment by jasonshelter0** [comment]
I built and released a plugin in a maintained fork in response to this request. It fixes the bug & improves performance, hopefully useful to people following the issue:

https://github.com/jasonshelter0/obsidian-extended-paste-mode

check this out via BRAT

**Comment by jglev** [comment]
Thank you all for your correspondence on this!

I've not been maintaining this (and several other plugins) regularly for quite a while; I'm sorry for my delay in responding! @jasonshelter0 , thank you for your work on this, and for the other optimizations you've made in the fork!

@jasonshelter0, am I correct in thinking that you haven't submitted the Extended-Paste-Mode fork to the official Obsidian plugin list at this point? Or is the fork submitted and just awaiting approval there? If the former, and if you were amenable to it, I'd be happy to merge the changes back into this codebase and add you to the contributors list and license file in this repo -- I do anticipate having more bandwidth going forward to get back into responsibly maintaining the codebase (especially thanks to Claude, CoPilot, etc., as you mentioned). Or if you are planning to maintain the fork over time, I'm open to deprecating this version and gracefully directing users towards the fork.

In either case, I'm sorry for the delay, but glad to be corresponding about it now!

**Comment by jasonshelter0** [comment]
Hello @jglev thanks for replying! I haven't submitted my fork since I'm currently use it for myself so feel free to merge my changes & mention me. I'm a student and a newcomer to the community so this would mean a lot to me! Use of AI in my fork is minimal and I've reviewed the code by myself so I think this should not be a problem. Thanks again & have a good day.

**Comment by jglev** [comment]
Thank you both again very much for taking the time to write this up!

I'm just commenting here, having also done so in #42, that I've released a new version of the plugin today that incorporates the changes from @jasonshelter0 (and made additional changes that are hopefully responsive to other open issues). If you would be willing to update the plugin from within Obsidian and see whether the changes do seem to address the issue; I'd be grateful! I'm going to close this issue for now in the hope that these changes are responsive to the writeup here; if you find that there are lingering issues, please feel free to comment back here, and we can reopen this issue.

@jasonshelter0 , again, thank you! I've added you to the License file and made a specific mention in the Readme file, as well.
